engine/pom.xml

```diff
@@ -105,6 +105,11 @@
       <artifactId>cron-utils</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
```

Comment:
:wrench: If we're going to go with the Guava one, I would strongly recommend [Caffeine](https://github.com/ben-manes/caffeine) instead, which is in general an improvement on the Guava caches (which is written by the original authors of the Guava cache I think). API should be pretty similar, so shouldn't be too hard to migrate. Just my two cents.

Reply:
Setting to 0 doesn't disable it. It's actually an invalid size. That makes me think, do we validate configuration option anywhere 🤔

Reply:
We have such validations normally in the config classes